databricks/koalas/indexing.py

```diff
@@ -492,21 +494,35 @@ def __setitem__(self, key, value):
         rows_sel, cols_sel = key
 
         if (not isinstance(rows_sel, slice)) or (rows_sel != slice(None)):
-            raise SparkPandasNotImplementedError(
-                description="""Can only assign value to the whole dataframe, the row index
-                has to be `slice(None)` or `:`""",
-                pandas_function=".loc[..., ...] = ...",
-                spark_target_function="withColumn, select")
-
-        if not isinstance(cols_sel, str):
-            raise ValueError("""only column names can be assigned""")
+            if isinstance(cols_sel, list):
+                kdf = self._kdf
+                is_start = True
+                for col_sel in cols_sel:
+                    if is_start:
+                        sdf = kdf._sdf.withColumn(
+                            col_sel,
+                            (F.when(F.col(kdf._internal.index_columns[0]).isin(rows_sel), value)
+                              .otherwise(F.col(col_sel))))
+                        is_start = False
+                    else:
+                        sdf = sdf.withColumn(
+                            col_sel,
+                            (F.when(F.col(kdf._internal.index_columns[0]).isin(rows_sel), value)
+                              .otherwise(F.col(col_sel))))
```

Comment:
How about:

```py
sdf = kdf._sdf
for col_sel in cols_sel:
    sdf = sdf.withColumn(
        col_sel,
        (F.when(F.col(kdf._internal.index_columns[0]).isin(rows_sel), value)
            .otherwise(F.col(col_sel))))
```

?

Reply:
it looks absolutely much better. Thanks !! 😄